### PR TITLE
NF: Paste clipboard image as PNG

### LINKED ISSUE #8709: Feature : Paste Clipboard Image As PNG
### **Is your feature request related to a problem? Please describe.**
AnkiDesktop have feature of "Paste Clipboard Image As PNG " in the preference window ,
currently AnkiDroid do support to copy and paste image but not have this preference option in app.

Capture

### PR DESCRIPTION
**Pull request by kartikeysaran**
## Fixes
Fixes #8709 

## How Has This Been Tested?
This has been tested on a real device

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -1598,7 +1598,12 @@ private String loadImageIntoCollection(Uri uri) throws IOException {
 
         Map.Entry<String, String> fileNameAndExtension = FileUtil.getFileNameAndExtension(filename);
 
-        File clipCopy = File.createTempFile(fileNameAndExtension.getKey(), fileNameAndExtension.getValue());
+        File clipCopy;
+        if (getCol().getConf().optBoolean("pastePNG")) {
+        clipCopy = File.createTempFile(fileNameAndExtension.getKey(), ".png");
+        } else {
+        clipCopy = File.createTempFile(fileNameAndExtension.getKey(), ".jpg");
+        }
```
[the remark this answers is not in the document]
Comment (kartikeysaran) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1606]: Can you please tell me which line and exact changes as i have followed the code style ? Thanks !

Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1606]:
```suggestion
        if (getCol().getConf().optBoolean("pastePNG")) {
            clipCopy = File.createTempFile(fileNameAndExtension.getKey(), ".png");
        } else {
            clipCopy = File.createTempFile(fileNameAndExtension.getKey(), ".jpg");
        }
```

File: AnkiDroid/src/main/res/xml/preferences_general.xml [hunk 2]
```diff
@@ -95,4 +96,4 @@
             android:key="widgetBlink"
             android:title="@string/notification_minimum_cards_due_blink" />
     </PreferenceCategory>
-</PreferenceScreen>
\ No newline at end of file
+</PreferenceScreen>
```
Comment (Akshay0701) [inline AnkiDroid/src/main/res/xml/preferences_general.xml]: can you revert this change?

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
Great! @kartikeysaran.

**Comment by mikehardy** [comment]
superceded by #9080

**Pull request closed without merging**